diff --git a/generic/default.nix b/generic/default.nix
--- a/generic/default.nix
+++ b/generic/default.nix
@@ -215,9 +215,12 @@
         export rundir="$(pwd)"
         ${lib.getBin test-driver}/bin/nixos-test-driver \
         ${lib.optionalString interactive "--interactive"} \
-        --start-scripts ${lib.concatStringsSep " " (nodes interactive)} \
+        --vm-names ${lib.concatStringsSep " " (lib.attrNames config.nodes)} \
+        --vm-start-scripts ${lib.concatStringsSep " " (nodes interactive)} \
+        --container-names \
+        --container-start-scripts \
           --vlans ${lib.concatStringsSep " " vlans} \
           -- ${hostPkgs.writeText "test-script" testScriptWithMounts}
       '';
